### PR TITLE
Triple set resolve fallback enable as default

### PR DESCRIPTION
**Pull request by AlbumenJ**
## What is the purpose of the change

1. Add `addIfAbsent` to prevent service with version to override service without version
2. Add logs
3. Set `RESOLVE_FALLBACK_TO_DEFAULT` as true by default

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by AlbumenJ** [comment]
@guohao @EarthChen PTAL

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/TripleProtocol.java [hunk 1]
```diff
@@ -113,9 +114,26 @@ public void afterUnExport() {
 
         invokers.add(invoker);
 
-        pathResolver.add(url.getServiceKey(), invoker);
+        Invoker<?> previous = pathResolver.add(url.getServiceKey(), invoker);
```
Comment (guohao) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/TripleProtocol.java:117]: 这里直接报错是不是更好些

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/TripleProtocol.java [hunk 2]
```diff
@@ -113,9 +114,26 @@ public void afterUnExport() {
...
+                + "], dubbo will override with invoker[" + url + "]";
+            if (url.getServiceKey().equals(url.getServiceModel().getServiceModel().getInterfaceName())) {
+                logger.info(msg);
+            } else {
+                logger.warn(LoggerCodeConstants.PROTOCOL_INCORRECT_PARAMETER_VALUES, "", "", msg);
+            }
+        }
         if (RESOLVE_FALLBACK_TO_DEFAULT) {
-            pathResolver.add(url.getServiceModel().getServiceModel().getInterfaceName(), invoker);
+            previous = pathResolver.addIfAbsent(url.getServiceModel().getServiceModel().getInterfaceName(), invoker);
```
Comment (guohao) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/TripleProtocol.java:128]: `addIfAbsent`的原因是什么？ 是否会依赖不同provider的初始化顺序

Comment (AlbumenJ) [reply dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/TripleProtocol.java:128]: `addIfAbsent` 是为了不让 fallback 的空版本号覆盖真正空版本号的服务

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/TripleProtocol.java (hunk 1 above)
Comment (AlbumenJ) [reply dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/TripleProtocol.java:116]: done

**Review by guohao — APPROVED** [review]

**Pull request merged**